### ISSUE TITLE
`Style/Encoding` cop doesn't detect magic comment if its not on the very beginning

### ISSUE THREAD

**Issue by ashmaroli** [bbatsov/rubocop#4986]
The `Style/Encoding` cop doesn't detect the magic comment `# Encoding: utf-8` if there is another magic comment above it.

--------

## Expected behavior

The Ruby docs does mention that the magic comment for Encoding be on the first line unless the first line contains a shebang comment. Therefore, expected the Encoding Cop to raise an offense against the magic comment for Encoding *after* another magic comment, as its futile.

## Actual behavior

The magic comment for Encoding on the second line is not flagged

## Steps to reproduce the problem

Run `rubocop` against `foo.rb` which is simply:
```ruby
# Frozen-string-literal: true
# Encoding: utf-8

def foo; end
```

## RuboCop version

```
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.3 i386-mingw32)
```

**Comment by bgeuken** [comment]
> The Ruby docs does mention that the magic comment for Encoding be on the first line unless the first line contains a shebang comment. 

It's actually the opposite. If there is a shebang, it should be on top.
That's also reflected in the [tests](https://github.com/bbatsov/rubocop/blob/master/spec/rubocop/cop/style/encoding_spec.rb#L33)  and the examples in the [style-guide](https://github.com/bbatsov/ruby-style-guide#magic-comments). 
Though the explanation in the style guide seems a bit misleading `Magic comments should only go below shebangs if they are needed in your source file.`.

**Comment by ashmaroli** [comment]
> It's actually the opposite. If there is a shebang, it should be on top.

@bgeuken That's what I said too.. **You** just **misread it..** :smiley: :
*"..the magic comment for Encoding be on the **first line unless the first line contains** a shebang comment."*
which is the same as:
*"..the magic comment for Encoding be on the **second line if the first line contains** a shebang comment."*

**Comment by bgeuken** [comment]
Seems I got you wrong. For what it's worth, the current encoding spec explicitly tests that the cop doesn't report an offense [when encoding is in the wrong place](https://github.com/bbatsov/rubocop/blob/master/spec/rubocop/cop/style/encoding_spec.rb#L58)

**Comment by ashmaroli** [comment]
> the current encoding spec explicitly tests that the cop doesn't report an offense when encoding is in the wrong place

Ah! I see.. Thanks for pulling that out for me. So, technically, this is by design.
In an ideal world, the presence of string `"# Encoding: utf-8\n\n"` anywhere would've been flagged.. just saying.. :smiley:
